launcher.py

```diff
@@ -230,144 +382,32 @@ def _get_install_commands_windows(self):
-
-        # Add MSYS2 paths to the PATH environment variable for the current session
-        logging.debug(
-            f"Addiing {msys2_root_path}\\usr\\bin, {msys2_root_path}\\mingw64\\bin, {msys2_root_path}\\mingw32\\bin to current enviornment."
-        )
-        current_path = os.environ.get("PATH", "")
-        new_path = (
-            f"{msys2_root_path}\\usr\\bin;{msys2_root_path}\\mingw64\\bin;{msys2_root_path}\\mingw32\\bin;"
-            + current_path
-        )
-        os.environ["PATH"] = new_path
-
-        logging.info("MSYS2 installed and updated successfully.")
-        logging.info("NOTE: Please restart your terminal before running this script again.")
-
-    except subprocess.CalledProcessError as e:
-        logging.error(f"Error installing MSYS2: {e}", exc_info=True)
-        sys.exit(1)
-    finally:
-        if Path.exists(installer_name):
-            os.remove(installer_name)
-
-
-def check_MediaProcessor(system):
-    if system == "Windows":
-        MediaProcessor_path = Path("MediaProcessor") / "build" / "MediaProcessor.exe"
-    else:
-        MediaProcessor_path = Path("MediaProcessor") / "build" / "MediaProcessor"
-    return MediaProcessor_path.exists()
-
-
-def generate_virtualenv():
-    logging.info(f"Generating virtual environment at: {VENV_DIR}")
-    run_command(f"{sys.executable} -m venv {str(VENV_DIR)} --system-site-packages")
-    logging.info("Successfully generated Virtual environment.")
-
-
-def install_python_dependencies(system):
-    try:
-        if not VENV_DIR.exists():
-            generate_virtualenv()
-
-        logging.info("Installing Python dependencies...")
-
-        if system == "Windows":
-            # Check for both `Scripts` and `bin` folders.
-            pip_scripts_path = VENV_DIR / "Scripts" / "pip.exe"
-            pip_bin_path = VENV_DIR / "bin" / "pip.exe"
-
-            if not pip_scripts_path.exists() and not pip_bin_path.exists():
-                logging.error(f"pip not found. Searched paths: [{pip_scripts_path}] and [{pip_bin_path}]")
-                sys.exit(1)
-
-            if pip_bin_path.exists():
-                pip_path = pip_bin_path
-            else:
-                pip_path = pip_scripts_path
-
-        else:
-            pip_path = VENV_DIR / "bin" / "pip"
-            if not pip_path.exists():
-                logging.error(f"pip not found. Searched paths: [{pip_path}]")
-                sys.exit(1)
-
-        run_command(f"{str(pip_path)} install -r requirements.txt")
-        logging.info("Python dependencies installed.")
-    except Exception as e:
-        logging.error(f"Failed to install Python dependencies: {e}", exc_info=True)
-        sys.exit(1)
+def ensure_MediaProcessor_build(system, re_build=False):
+    """Ensure that the MediaProcessor is build. If rebuild is true, it will rebuild the MediaProcessor."""
+    if not MEDIAPROCESSOR_PATH.exists():
+        os.makedirs(MEDIAPROCESSOR_PATH)
```

Comment:
don't like two things here: first we're explicit about the name of the C++ binary, which could change and no reason to couple that with the backend. Second, we're not following naming convention due to the `MediaProcessor`

Please name this to `build_processing_engine()`

Reply:
Should i change `MEDIAPROCESSOR_PATH` to `MEDIAPROCESSOR_BUILD_DIRECTORY` or `PRCOESSING_ENGINE_DIRECTORY`?

Reply:
> Should i change `MEDIAPROCESSOR_PATH` to `MEDIAPROCESSOR_BUILD_DIRECTORY` or `PRCOESSING_ENGINE_DIRECTORY`?

Let's go with `PROCESSING_ENGINE_PATH`. Could you please check in general for consistency. 
I don't mind calling something a directory, but if the variable is holding literally a path, in that case we should call it a path.